### PR TITLE
roles: hosted_engine_setup: Add pause option before running engine-setup

### PR DESCRIPTION
**Pull request by arachmani**
Add the option to pause the deployment and adjust changes in the
bootstrap engine VM before running the engine-setup

Bug-Url: https://bugzilla.redhat.com/1959273
Signed-off-by: Asaf Rachmani <arachman@redhat.com>

### CODE REVIEW

File: roles/hosted_engine_setup/tasks/bootstrap_local_vm/03_engine_initial_tasks.yml [hunk 1]
```diff
@@ -54,16 +54,24 @@
       - "OVESETUP_DWH_DB/password=str:{{ he_dwh_db_password }}"
     when: he_dwh_db_password is defined
   - name: Import OpenSCAP task
-    import_tasks: apply_openscap_profile.yml
+    import_tasks: ../apply_openscap_profile.yml
```
Comment (didib) [inline roles/hosted_engine_setup/tasks/bootstrap_local_vm/03_engine_initial_tasks.yml:57]: Why do we need to import from "../"? And is it only in this file? Perhaps worth a separate patch.

File: roles/hosted_engine_setup/tasks/bootstrap_local_vm/03_engine_initial_tasks.yml [hunk 2]
```diff
@@ -54,16 +54,24 @@
...
     include_tasks: "{{ item }}"
     with_fileglob: "hooks/enginevm_before_engine_setup/*.yml"
     register: include_before_engine_setup_results
-  - debug: var=include_before_engine_setup_results
+  - name: Pause the execution to allow the user to configure the bootstrap engine VM
+    block:
+      - name: Allow the user to connect to the bootstrap engine and change configuration
+        debug:
+          msg: >-
+            You can now connect to the bootstrap engine using the temporary IP address -
```
Comment (didib) [inline roles/hosted_engine_setup/tasks/bootstrap_local_vm/03_engine_initial_tasks.yml:68]: Perhaps 'using ssh as root, from this host, to the temporary...'. I suppose it's only from this host, right?

**Comment by didib** [comment]
Also, perhaps add a Bug-Url: - perhaps https://bugzilla.redhat.com/1956413 .

Also, perhaps pause (optionally, but true by default) after engine-setup, if it fails, like he_pause_after_failed_add_host .

File: roles/hosted_engine_setup/tasks/bootstrap_local_vm/03_engine_initial_tasks.yml (hunk 1 above)
Comment (arachmani) [reply roles/hosted_engine_setup/tasks/bootstrap_local_vm/03_engine_initial_tasks.yml:57]: This is handled in https://github.com/oVirt/ovirt-ansible-collection/pull/271, so we need to rebase it.

Comment (didib) [reply roles/hosted_engine_setup/tasks/bootstrap_local_vm/03_engine_initial_tasks.yml:57]: I do not see that you rebased. Not sure about your exact plans nor whether github makes it easy to rebase a PR on top of another PR.

**Comment by didib** [comment]
> > Also, perhaps add a Bug-Url: - perhaps https://bugzilla.redhat.com/1956413 .
> > Also, perhaps pause (optionally, but true by default) after engine-setup, if it fails, like he_pause_after_failed_add_host .
> 
> I've opened a bug and added the Bug-Url, I prefer to add the `he_pause_after_failed...` in a different PR.

I see Bug-Url only in the PR, not on any specific patch - and AFAICT, won't make it to the git log.

**Comment by arachmani** [comment]
> I see Bug-Url only in the PR, not on any specific patch - and AFAICT, won't make it to the git log.

Done, added to the last commit.

**Comment by didib** [comment]
For some reason it has "Bug-Url: Bug-Url:", twice. Not sure our tooling will parse this correctly.
I'd also personally just squash everything to a single patch, I do not see much value in merging these multiple patches, but don't mind that much.

**Comment by arachmani** [comment]
> For some reason it has "Bug-Url: Bug-Url:", twice. Not sure our tooling will parse this correctly.
> I'd also personally just squash everything to a single patch, I do not see much value in merging these multiple patches, but don't mind that much.

Thanks, fixed the double Bug-Url.
We will squash it to a single commit when merging.

**Pull request merged**